src/redux/rockets/rocketSlice.jsx

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable array-callback-return */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const url = 'https://api.spacexdata.com/v3/rockets';
+
+const initialState = {
+  rockets: [],
+  isLoading: false,
+};
+
+export const getRockets = createAsyncThunk('book/getBooks', async (_, thunkAPI) => {
+  try {
+    const response = await fetch(url);
+    return await response.json();
+  } catch (error) {
+    return thunkAPI.rejectWithValue(error);
+  }
+});
+
+const rocketSlice = createSlice({
+  name: 'rocket',
+  initialState,
+  reducers: {
+    reserveRocket: (state, action) => {
+      const newRocket = state.rockets.map((rocket) => {
+        if (rocket.id !== action.payload) return rocket;
+        return { ...rocket, reserved: true };
+      });
+      return {
+        ...state,
+        rockets: newRocket,
+      };
+    },
+    leaveRocket: (state, action) => {
+      const newRocket = state.rockets.map((rocket) => {
+        if (rocket.id !== action.payload) return rocket;
+        return { ...rocket, reserved: false };
+      });
+      return {
+        ...state,
+        rockets: newRocket,
+      };
+    },
+
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(getRockets.pending, (state) => ({
+        ...state,
+        isLoading: true,
+      }))
+      // .addCase(getRockets.fulfilled, (state, action) => {
+      //   console.log(action.payload);
+      // })
```

Comment:
Kindly remove all commented-out code pieces of code from your project to keep your codebase clean🙏

Reply:
Sure, we'll do that as well.